### DISCUSSION TITLE
Simple countdown timer for a daily notes template

### DISCUSSION THREAD

**Discussion by axelson** [SilentVoid13/Templater#722] [Templates Showcase]
Here's a simple countdown template:
```
<%*
  const elixirConfDate = new Date('2022-08-30');
  const now = new Date();
  const diff = elixirConfDate - now;
  const days = Math.floor(diff / (1000 * 3600 * 24));
%><% days %> days until my event!
```

If this is used to create a daily template on 2022-08-13 then it will print

    16 days until my event!

**Comment by ricardonevesmusic** [comment]
Hi

Could you try to make a prompt version?

It would be easier to insert it on the document with Templater.

I've tried something here, but it didn't work out very well:

```
<%*
  tp.system.prompt(new Date());
  const elixirConfDate = new Date('');
  const now = new Date();
  const diff = elixirConfDate - now;
  const days = Math.floor(diff / (1000 * 3600 * 24));
%><% days %> days until my event!

```
If you could create a prompt version, that would make life easier.

Then we could just write the date or pick from the calendar.

Writing the date alone, would be enough.

Yeah, that's it.

That way we could have 2 different versions of the same idea/concept.

**Reply by talraash** [reply]
If you still need... You can little modify code for get HH:mm:ss in output

```js
<%*
const defValue = await new Date().toISOString().split('T')[0]; // Get current date as str to default value for prompt 
const promtDate = await tp.system.prompt('Write new date in "YYYY-MM-DD" fotrmat', defValue, true);
const elixirConfDate = await new Date(promtDate);
const now = await new Date();
const diff = await elixirConfDate - now;
const days = await Math.floor(diff / (1000 * 3600 * 24));
%><% days %> days until my event!
```

**Reply by ricardonevesmusic** [reply]
Nice.

Thank you.

I'm not using Obsidian at the moment (started using some simpler tools like Google Calendar and Google Keep, trying to implement GTD), but if I start using it again, I'll take this template in consideration.

Thank you.

Hope you have a nice day! 👍

**Comment by bhuvang14** [comment]
> Here's a simple countdown template:
> 
> ```
> <%*
>   const elixirConfDate = new Date('2022-08-30');
>   const now = new Date();
>   const diff = elixirConfDate - now;
>   const days = Math.floor(diff / (1000 * 3600 * 24));
> %><% days %> days until my event!
> ```
> 
> If this is used to create a daily template on 2022-08-13 then it will print
> 
> ```
> 16 days until my event!
> ```

how  can make this work using templater

**Comment by arturblch** [comment]
One more realisation
`$= dv.date("<% tp.date.now("YYYY-MM-DDTHH:mm") %>").toRelative()`
